docs/_user_guide/best_practices_optimization.md

```diff
@@ -1,306 +1,354 @@
+
 Type 'q()' to quit R.
+>
+```
 
-> library('gurobi')
+```title="Terminal Command"
+library('gurobi')
+```
+```{ .yaml .no-copy title="Terminal Output" }
 Loading required package: slam
 >
 ```
 
 You can now run the R scripts to solve the optimization problem using Gurobi on interactive Yen nodes.
 
-**Important:** You need to load the `gurobi` and `R` modules every time you login to the Yens
-before running the interactive R scripts that use `gurobi` R package.
+!!! Warning
+    The default Gurobi 10 version is not compatible with the default R 4.3 version. To ensure proper functionality, you need to use R 4.2 instead. Running `ml gurobi R` with an incompatible version will result in an error. For more details on which Gurobi versions work with specific R versions, please refer to the information provided [here](https://support.gurobi.com/hc/en-us/articles/360025593712-Which-R-versions-are-supported-by-Gurobi). 
 
-Note, that if you want to use an older Gurobi version, you will need to install `gurobi` R package to your user
-R library. 
+!!! Important
+    You need to load the `gurobi` and `R` modules every time you log in to the Yens before running the interactive R scripts that use `gurobi` R package.
+
+Note that if you want to use an older Gurobi version, you will need to install `gurobi` R package to your user R library.
 
 Load R and the older Gurobi module:
 
+```title="Terminal Command"
+module purge
+ml R/4.2 gurobi/9.5.2
 ```
-$ module purge
-$ ml R gurobi/9.5.2
+
+Start R by typing `R`:
+
+```title="Terminal Command"
+R
 ```
 
-Start R by typing `R`, then install a personal `gurobi` package that links to Gurobi 9:
+Then install a personal `gurobi` package that links to Gurobi 9:
 
-```R
+```title="Terminal Command"
+install.packages("/software/non-free/Gurobi/gurobi952/linux64/R/gurobi_9.5-2_R_4.2.0.tar.gz", repos=NULL)
+```
+```{ .yaml .no-copy title="Terminal Output" }
 > install.packages("/software/non-free/Gurobi/gurobi952/linux64/R/gurobi_9.5-2_R_4.2.0.tar.gz", repos=NULL)
-Installing package into ‘/home/users/$USER/R/x86_64-pc-linux-gnu-library/4.2’
-(as ‘lib’ is unspecified)
+Warning in install.packages("/software/non-free/Gurobi/gurobi952/linux64/R/gurobi_9.5-2_R_4.2.0.tar.gz",  :
+  'lib = "/software/free/R/R-4.2.1/lib/R/library"' is not writable
+Would you like to use a personal library instead? (yes/No/cancel) yes
+Would you like to create a personal library
+‘/zfs/home/users/wolee/R/x86_64-pc-linux-gnu-library/4.2’
+to install packages into? (yes/No/cancel) yes
 * installing *binary* package ‘gurobi’ ...
 * DONE (gurobi)
-> library("gurobi")
+```
+
+```title="Terminal Command"
+library("gurobi")
+```
+```{ .yaml .no-copy title="Terminal Output" }
 Loading required package: slam
 >
 ```
 
 Quit R. You can now run the R scripts to solve the optimization problem using Gurobi on interactive Yen nodes.
 
 
-#### AMPL with Knitro Solver
-### Running AMPL with Knitro Solver in Python
+### AMPL with Knitro Solver
+
```

Comment:
I would maybe add a warning about only one floating license for AMPL

Reply:
Added it to the existing warning. Does this look good to you?

To ensure that the AMPL instance is always properly released (even in cases of errors or exceptions), we must properly close `ampl` instance. The `try` and `finally` logic does that in which we release the AMPL license with `ampl.close()` call. Always shut down the Jupyter kernel since there is only one floating license for AMPL.

Reply:
Added

Reply:
that looks fine